### PR TITLE
HSEARCH-3353 Restore support for search hit explanation

### PR DESCRIPTION
**Pull request by yrodiere**
https://hibernate.atlassian.net//browse/HSEARCH-3353

~Based on #1971 , which should be merged first.~ => Done

See the tests added to `ElasticsearchExtensionIT` and `LuceneExtensionIT` for how the feature will be used. It's a bit different from what we used to provide through `org.hibernate.search.jpa.FullTextQuery#explain`, but I think it is more convenient.

### CODE REVIEW

File: backend/elasticsearch/src/main/java/org/hibernate/search/backend/elasticsearch/search/query/ElasticsearchSearchQuery.java [hunk 1]
```diff
@@ -9,4 +9,30 @@
...
+	 *
+	 * @param id The id of the document to test.
+	 * @return A string representing a JSON object describing the score computation for the hit.
+	 * @throws org.hibernate.search.util.common.SearchException If the query targets multiple indexes,
+	 * or if the explain request fails.
+	 */
+	String explain(String id);
+
+	/**
+	 * Explain score computation of this query the document with the given id in the given index.
```
Comment (fax4ever) [inline backend/elasticsearch/src/main/java/org/hibernate/search/backend/elasticsearch/search/query/ElasticsearchSearchQuery.java]: maybe to the document?

File: backend/lucene/src/main/java/org/hibernate/search/backend/lucene/search/query/LuceneSearchQuery.java [hunk 2]
```diff
@@ -8,5 +8,32 @@
...
+	 *
+	 * @param id The id of the document to test.
+	 * @return An {@link org.apache.lucene.search.Explanation} describing the score computation for the hit.
+	 * @throws org.hibernate.search.util.common.SearchException If the query targets multiple indexes,
+	 * or if the explain request fails.
+	 */
+	Explanation explain(String id);
+
+	/**
+	 * Explain score computation of this query the document with the given id in the given index.
```
Comment (fax4ever) [inline backend/lucene/src/main/java/org/hibernate/search/backend/lucene/search/query/LuceneSearchQuery.java]: I wouldn't know... query to the document maybe?

**Comment by yrodiere** [comment]
Thanks; I fixed the javadoc and merged.

**Comment by fax4ever** [comment]
Thank you!

**Pull request merged**